wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -3,4 +3,106 @@
  * Add/update/retrieve Tapestry post and its child nodes
  * 
  */
-class TapestryController {}
\ No newline at end of file
+
+class TapestryController {
+    const POST_TYPES = [
+        'TAPESTRY' => 'tapestry',
+        'TAPESTRY_NODE' => 'tapestry_node'
+    ];
+    const ERRORS = [
+        'INVALID_POST_ID' => [
+            'MESSAGE' => 'PostID is invalid',
+            'STATUS' => ['status' => 404]
+        ]
+    ];
+    private $postId;
+
+    /**
+     * Constructor
+     */
+    public function __construct($postId) {
+        $this->postId = $postId;
```

Comment:
You should validate `$postId` here to make sure it is a number, that it exists, and that it is a tapestry post type.